src/parser/jobs/ast/modules/SleeveDraw.tsx

```diff
@@ -0,0 +1,88 @@
+import {Trans} from '@lingui/react'
+import {ActionLink, StatusLink} from 'components/ui/DbLink'
+import {getDataBy} from 'data'
+import ACTIONS from 'data/ACTIONS'
+import STATUSES, {Status} from 'data/STATUSES'
+import {BuffEvent, CastEvent} from 'fflogs'
+import Module, {dependency} from 'parser/core/Module'
+import Checklist from 'parser/core/modules/Checklist'
+import Suggestions, {SEVERITY, TieredSuggestion} from 'parser/core/modules/Suggestions'
+import React from 'react'
+import Sect from './Sect'
+
+const SEVERITY_MOD = {
+	MINOR: 0.1,
+	MEDIUM: 0.3,
+	MAJOR: 0.5,
+}
+
+// Lifted from WHM benison and adapted to AST and TSX
+export default class SleeveDraw extends Module {
+	static handle = 'SleeveDraw'
+
+	@dependency private suggestions!: Suggestions
+	@dependency private checklist!: Checklist
+
+	private lastUse = 0
+	private uses = 0
+	private totalHeld = 0
+
+	private activeSect: Status | undefined
+
+	protected init() {
+		this.addEventHook('cast', {abilityId: ACTIONS.CELESTIAL_OPPOSITION.id, by: 'player'}, this.onCast)
+		this.addEventHook('applybuff', {abilityId: [STATUSES.DIURNAL_SECT.id, STATUSES.NOCTURNAL_SECT.id], by: 'player'}, this.onSect)
+		this.addEventHook('complete', this.onComplete)
+	}
+
+	private onCast(event: CastEvent) {
+		this.uses++
+		if (this.lastUse === 0) { this.lastUse = this.parser.fight.start_time }
+
+		const held = event.timestamp - this.lastUse - (ACTIONS.CELESTIAL_OPPOSITION.cooldown * 1000)
+		if (held > 0) {
+			this.totalHeld += held
+		}
```

Comment:
This is redundant, unless held can go negative. `this.totalHeld += held` where held is 0 will just add zero. Neater to clamp the value (if needed) and just yolo add it imo, but more a question of preference I guess.